doc: stdcm: add documentation (#72)
DGEXSolutions/osrd-website: content/docs/reference/design-docs/stdcm/pathfinding_module/conflict_avoidance/index.en.md

---
title: "Conflict avoidance"
linkTitle: "4 - Conflict avoidance"
weight: 40
---

> While exploring the graph, it is possible to end up in places that would
> cause conflicts. They can be avoided by adding delay.

Reviewer: Just a quick proposal to avoid repeating "cause conflicts" too many times. Feel free to keep the sentence you prefer though.
```suggestion
While exploring the graph, it is possible to end up in locations that would
generate conflicts. They can be avoided by adding delay.
```

Once a path is found, the value is summed over the whole path.
This is added to the departure time.

> For example :
> - a train leaves between 10:00 and 11:00. The initial maximum
> time shift is 1:00.
> - At some point, an edge becomes unavailable 20 minutes after the
> train passage. The value is now at 20 for any edge accessed from here.
> - The departure time is then delayed by 5 minutes to avoid a conflict.
> The maximum time shift value is now at 15 minutes.
> - This process is applied until the destination is found,
> or until no more delay can be added this way.

#### Engineering allowances

Once the maximum delay is at 0, the delay needs to be added
between two points of the path.

![Engineering allowances (1/2)](engineering_allowance.png)

The idea is the same as the one used to fix speed discontinuities:
new edges are created, replacing the previous ones.
The new edges have an engineering allowance, to add the delay where
it is possible.

![Engineering allowances (2/2)](engineering_allowance_edges.png)

computing an
[engineering allowance]({{< ref "docs/explanation/running_time_calculation/allowances" >}} "allowances")

> is a feature of the run time

Reviewer: ```suggestion
is a feature of the running-time
```

---
title: "Conflict avoidance"
linkTitle: "4 - Conflict avoidance"
weight: 40
---

> While exploring the graph, it is possible to end up in places that would
> cause conflicts. They can be avoided by adding delay. This page
> describes how this is done.

Reviewer 2: ```suggestion
While exploring the graph, it is possible to end up in places that would
cause conflicts. They can be avoided by adding delay.
```

#### Shifting the departure time

> In the module parameters, the departure time is defined as an interval:

Reviewer 2: ```suggestion
The departure time is defined as an interval in the module parameters:
```
